MaxText/convert_deepseek_ckpt.py

```diff
@@ -0,0 +1,517 @@
+"""
+Copyright 2025 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+r"""Convert weights from a DeepSeek style model to a MaxText one.
+
+Example cmd:
+
+python3 MaxText/convert_deepseek_ckpt.py --base_model_path <path/to/meta/ckpt> \
+    --maxtext_model_path <GCS/path/to/save/new/maxtext/ckpt> --model_size deepseek2-16b
+"""
+
+# pylint: disable=line-too-long
+# pylint: disable=unsupported-assignment-operation
+# pytype: disable=unsupported-operands
+
+import argparse
+import pathlib
+import os
+import gc
+import logging
+
+import numpy as np
+import torch
+import psutil
+from tqdm import tqdm
+
+import max_logging
+from safetensors import safe_open
+import max_utils
+
+
+MODEL_PARAMS_DICT = {
+    "deepseek2-16b": {
+        "num_layers": 27,
+        "first_num_dense_layers": 1,
+        "base_num_query_heads": 16,
+        "base_emb_dim": 2048,
+        "num_experts": 64,
+        "q_lora_rank": 0,
+        "kv_lora_rank": 512,
+        "qk_nope_head_dim": 128,
+        "qk_rope_head_dim": 64,
+        "v_head_dim": 128,
+    },
+    "deepseek3-671b": {
+        "num_layers": 61,
+        "first_num_dense_layers": 3,
+        "base_num_query_heads": 128,
+        "base_emb_dim": 7168,
+        "num_experts": 256,
+        "q_lora_rank": 1536,
+        "kv_lora_rank": 512,
+        "qk_nope_head_dim": 128,
+        "qk_rope_head_dim": 64,
+        "v_head_dim": 128,
+    },
+}
+
+
+# skip to convert ckpts related to fp8 and MTP for now
+SKIPPED_KEYS_SUFFIX = [
+    "_scale_inv",
```

Comment:
I don't think we need this `_scale_inv` here, as this script don't work on fp8 checkpoint and for bf16 converted checkpoint `_scale_inv` .

It will be better if we removed `_scale_inv` check, so it will fail if we run on fp8 checkpoint (even better, if you could check "_scale_inv" suffix and provide an error message that the script shouldn't be run on fp8 checkpoint.

If you remove "_scale_inv" from the list, the rest refers to MTP, so perhaps you can update the name SKIPPED_KEYS_SUFFIX to MTP_KEYS_SUFFIX

Reply:
Sounds good! I will make the update.

Reply:
Updated.